### PR TITLE
Added tensor layer numbers

### PR DESCRIPTION
**Pull request by dkun7944**
https://github.com/ggerganov/llama.cpp/issues/1493

Added a tensor layer number parameter to `ggml_tensor`. 0 is used for global tensors

### CODE REVIEW

File: ggml.h [hunk 1]
```diff
@@ -512,6 +513,7 @@ extern "C" {
 
     GGML_API const char * ggml_get_name(const struct ggml_tensor * tensor);
     GGML_API void         ggml_set_name(struct ggml_tensor * tensor, const char * name);
+GGML_API void         ggml_set_layer_num(struct ggml_tensor * tensor, int layer_num);
```
Comment (ProgramComputer) [inline ggml.h:516]: indent here

**Review by ProgramComputer — APPROVED** [review]

**Comment by ggerganov** [comment]
See my latest comment: https://github.com/ggerganov/llama.cpp/issues/1493#issuecomment-1585536451

**Comment by alitariq4589** [comment]
Can one of the admins verify this patch?

**Comment by mofosyne** [comment]
obsolete?

**Pull request closed without merging**